src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferConstCharOverConstUnitStringAnalyzer.cs

```diff
@@ -0,0 +1,105 @@
+                                                                                      isPortedFxCopRule: false,
+                                                                                      isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterCompilationStartAction(compilationContext =>
+            {
+                // Check that the object is a StringBuilder
+                if (!compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextStringBuilder, out INamedTypeSymbol? stringBuilderType))
+                {
+                    return;
+                }
+
+                IMethodSymbol appendStringMethod = stringBuilderType
+                    .GetMembers("Append")
+                    .OfType<IMethodSymbol>()
+                    .WhereAsArray(s =>
+                        s.Parameters.Length == 1 &&
+                        s.Parameters[0].Type.SpecialType == SpecialType.System_String)
+                    .FirstOrDefault();
+                if (appendStringMethod is null)
+                {
+                    return;
+                }
+
+                compilationContext.RegisterOperationAction(operationContext =>
+                {
+                    var invocationOperation = (IInvocationOperation)operationContext.Operation;
+                    if (invocationOperation.Arguments.Length < 1)
+                    {
+                        return;
+                    }
+
+
+                    if (!invocationOperation.TargetMethod.Equals(appendStringMethod))
+                    {
+                        return;
+                    }
+
+                    ImmutableArray<IArgumentOperation> arguments = invocationOperation.Arguments;
+                    IArgumentOperation firstArgument = arguments[0];
+
+                    // We don't handle class fields. Only local declarations
+                    if (!IsConstVar(firstArgument))
+                    {
+                        return;
+                    }
+
+                    if (firstArgument.Value is ILocalReferenceOperation argumentValue)
+                    {
+                        operationContext.ReportDiagnostic(argumentValue.CreateDiagnostic(Rule));
+                    }
+                    else if (firstArgument.Value is ILiteralOperation literalArgumentValue)
+                    {
+                        operationContext.ReportDiagnostic(literalArgumentValue.CreateDiagnostic(Rule));
+                    }
+                }
+                , OperationKind.Invocation);
+            });
+        }
+
+        private static bool IsConstVar(IArgumentOperation argument)
```

Comment:
[cosmetic][suggestion] Could be rewritten as a single return.

```csharp
=> argument.Value.ConstantValue.HasValue && argument.Value.ConstantValue.Value is string unitString && unitString.Length == 1
```

Reply:
Agreed, also if this is a private static method, can we make it a static local function?